src/repr/src/adt/datetime.rs

```diff
@@ -787,6 +787,11 @@ impl ParsedDateTime {
         Ok(())
     }
     pub fn check_datelike_bounds(&mut self) -> Result<(), String> {
+        if let Some(year) = self.year {
+            if year.unit == 0 {
+                return Err(format!("YEAR cannot be zero"));
```

Comment:
nit, since you're not actually interpolating any variables in `format!` using `"foo".to_string()` or `"foo".into()` is more idiomatic